staging/src/k8s.io/api/authentication/v1/types.go

```diff
@@ -105,3 +106,44 @@ type ExtraValue []string
 func (t ExtraValue) String() string {
 	return fmt.Sprintf("%v", []string(t))
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TokenRequest requests a token for a given service account.
```

Comment:
To jordans comment, I do think this needs to either be in v1alpha1, or v2alpha1 (technically it could be v1alpha1 since this won't cause us to rev v1 authentication)

Reply:
this is tricky, since we're wanting a subresource in api/v1... whatever surfaces there sticks forever (hence replicationcontrollers/scale being stuck at extensions/v1beta1 Scale forever)

Reply:
What @liggitt said. I think the best approach is to use `authentication.k8s.io/v1` and be pretty confident (e.g. by keeping the API minimal) when we move this feature from alpha to beta. We can then begin to move serviceaccount out of the main API group into `authentication.k8s.io/v2alpha1` which will give us a chance to correct any problems we encounter.

Reply:
The old "shrug, we can't solve a hard problem so we'll do the thing we tell people not to do" approach?

I agree the subresource is hard.  Maybe we should just call the subresource `alphatokenrequest` to force it to be clear, and only promote once it's beta (to `tokenrequest`).

Reply:
Even if we make an `/alphatoken` we can't support two versions without creating a new version of this API group and leaving v1 sparse. The `/token` subresource will be alpha feature guarded so having an `/alphatoken` path during alpha doesn't add much value unless `/token` accepts a separate type and both are served simultaneously.

Reply:
Since it's behind a gate we can punt on this.